### PR TITLE
make itertools.chain be a class and from_iterable a class_method of c…

### PR DESCRIPTION
**Pull request by IuryAlves**
In python `itertools.chain` is implemented as a class and `from_iterable` is a classmethod.

### CODE REVIEW

**Comment by trotterdylan** [comment]
Thanks for the PR! The test is failing because neither classmethod nor decorators are implemented yet. Decorators is a bit involved but I think classmethods aren't too bad, and once we have them this change can be made to work this way:

```
def from_iterable(...):
  ...
from_iterable = classmethod(from_iterable)
```

I've filed issues https://github.com/google/grumpy/issues/72 and https://github.com/google/grumpy/issues/73

**Comment by IuryAlves** [comment]
Thanks @trotterdylan

**Comment by trotterdylan** [comment]
classmethod is now supported so this PR could work if you replace `@classmethod` with `from_iterable = classmethod(from_iterable)`

File: lib/itertools.py [hunk 1]
```diff
@@ -17,15 +17,16 @@
 import _collections
 import sys
 
+def _from_iterable(cls, iterable):
```
Comment (S-YOU) [inline lib/itertools.py]: This two lines supposed to be inside chain class, I think. 
Something like 
https://docs.python.org/2/whatsnew/2.4.html#pep-318-decorators-for-functions-and-methods

Any issue with inside the class?

Comment (IuryAlves) [inline lib/itertools.py]: Nope. I just forgot that i could implement `from_iterable` inside the class.

**Comment by kit1978** [comment]
> @@ -17,15 +17,16 @@
 import _collections
 import sys
 
+def _from_iterable(cls, iterable):This two lines supposed to be inside chain class, I think.
Something like

**Comment by kit1978** [comment]
ขอ⁣ขอบคุณพี่ๆทุกท่านที่ช่วยเหลือครับ

ส่งจาก BlueMail ​

ใน 11 ม.ค. 2017, 20:22 ที่ 20:22 YOU <notifications@github.com> เขียน:

**Comment by trotterdylan** [comment]
Looks great. Thanks for getting this done!

**Pull request merged**